assets/js/modules/analytics/components/wp-dashboard/WPAnalyticsDashboardWidgetOverview.js

```diff
@@ -106,7 +143,33 @@ export default withData(
 			data: overviewReportDataDefaults,
 			priority: 1,
 			maxAge: getTimeInSeconds( 'day' ),
-			context: [ 'WPDashboard' ],
+			context: 'WPDashboard',
+			toState( state, { data } ) {
+				if ( ! state.overview ) {
+					return {
+						overview: calculateOverviewData( data ),
+					};
+				}
+			},
+		},
+		{
+			type: TYPE_MODULES,
+			identifier: 'analytics',
+			datapoint: 'report',
+			data: userReportDataDefaults,
+			priority: 1,
+			maxAge: getTimeInSeconds( 'day' ),
+			context: 'WPDashboard',
+			toState( state, { data } ) {
+				if ( ! state.directTotalUsers ) {
+					const directTotalUsers = get( data, '[0].data.totals[0].values[0]' );
+					const previousTotalUsers = get( data, '[0].data.totals[1].values[0]' );
+					return {
+						directTotalUsers,
+						previousTotalUsers,
+					};
```

Comment:
Let's use conditional chaining here too since it's all new. Also, (perhaps a bit of a stylistic nitpick) it might be a bit cleaner to include the values in the definition of the returned object rather than initializing variables right before the return. i.e.` directTotalUsers: data?.[0]?.data?.totals?.[0]?.values?.[0],`

If these values are accessed in multiple places repeatedly, it might be worth extracting a little `parseTotalUsersData` utility.

Reply:
Adjusted.

Reply:
I fixed this instance. I like your `parseTotalUsersData` idea, we currently use this in four places so seems useful. I will add that.